catalog/harvest/catalog-harvest-manager/src/main/java/org/codice/ddf/catalog/harvest/manager/HarvesterManager.java

```diff
@@ -0,0 +1,225 @@
+      throw new IllegalArgumentException("Property {monitoredLocation} may not be empty or null");
+    }
+
+    monitoredLocation = stripEndingSlash(monitoredLocation);
+
+    switch (processingMechanism) {
+      case DELETE:
+      case MOVE:
+        contentDirectoryMonitor = new ContentDirectoryMonitor(camelContext);
+        contentDirectoryMonitor.setMonitoredDirectoryPath(monitoredLocation);
+        contentDirectoryMonitor.setProcessingMechanism(processingMechanism);
+        contentDirectoryMonitor.setNumThreads(numThreads);
+        contentDirectoryMonitor.setReadLockIntervalMilliseconds(readLockIntervalMilliseconds);
+        contentDirectoryMonitor.setAttributeOverrides(getOverridesAsList());
+        contentDirectoryMonitor.init();
+        break;
+      case IN_PLACE:
+        createInPlaceHarvester(monitoredLocation);
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "Received invalid processing mechanism [%s], but expected [%s].",
+                processingMechanism, SUPPORTED_MECHANISMS));
+    }
+  }
+
+  private List<String> getOverridesAsList() {
+    return attributeOverrides
+        .entrySet()
+        .stream()
+        .map(entry -> entry.getKey() + "=" + entry.getValue())
+        .collect(Collectors.toList());
+  }
+
+  private void createInPlaceHarvester(String location) {
+    if (monitoredLocation.startsWith(HTTP)) {
+      webDavHarvester =
+          new WebDavHarvester(
+              location,
+              Collections.singleton(new PersistentListener(metacardOnlyAdapter, location)));
+    } else {
+      inPlaceDirectoryHarvester =
+          new DirectoryHarvester(
+              location,
+              Collections.singleton(new PersistentListener(metacardOnlyAdapter, location)));
+    }
+  }
+
+  /**
+   * Invoked when updates are made to the configuration of existing content monitors. This method is
+   * invoked by the container as specified by the update-strategy and update-method attributes in
+   * Blueprint XML file.
+   *
+   * @param properties - properties map for the configuration
+   */
+  public void updateCallback(Map<String, Object> properties) {
+    if (MapUtils.isNotEmpty(properties)) {
+      setProcessingMechanism(getPropertyAs(properties, "processingMechanism", String.class));
+      setMonitoredLocation(getPropertyAs(properties, "monitoredLocation", String.class));
+      setNumThreads(getPropertyAs(properties, "numThreads", Integer.class));
+      setReadLockIntervalMilliseconds(
+          getPropertyAs(properties, "readLockIntervalMilliseconds", Integer.class));
+
+      Object o = properties.get(Constants.ATTRIBUTE_OVERRIDES_KEY);
+      if (o instanceof String[]) {
+        String[] incomingAttrOverrides = (String[]) o;
+        setAttributeOverrides(Arrays.asList(incomingAttrOverrides));
+      }
+
+      destroy(0);
+      init();
+    }
+  }
+
+  public void destroy(int code) {
```

Comment:
✏️ Can the `code` argument be removed?

Reply:
Unfortunately no. The MSF `destroy-method` takes this as an argument. If it is not here, then there are odd behaviors on when this method is called. So far I know that code=1 when the configuration is deleted, and code=2 when the system is being shutdown.

Reply:
I think requiring a code is actually a bug : https://issues.apache.org/jira/browse/ARIES-1436.  Upgrading CM would eliminate the need for this argument

Reply:
I added a suppression for the SonarQube code smell.